File: v6/prolog/compile/test/typegen_golden/generic_expansion_end_to_end.types.rs
#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
#[serde(tag = "tag", content = "value", rename_all = "snake_case")]
pub enum DlOption<T> {
    None,
    Some(T),
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct AuthorAudit {
    pub id: i64,
    pub tag: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Bucket {
    pub id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Metric {
    pub id: i64,
    pub value: DlOption<i64>,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct MetricCopy {
    pub id: i64,
    pub value: DlOption<i64>,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Person {
    pub id: i64,
    pub name: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct PriorityHigh {
    pub id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct PriorityLow {
    pub id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct PriorityTag {
    pub id: i64,
    pub tag: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Review {
    pub id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct SeenBucket {
    pub id: i64,
    pub list_id: i64,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct SeenReview {
    pub id: i64,
    pub name: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct SeenTicket {
    pub id: i64,
    pub tag: String,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct Ticket {
    pub id: i64,
    pub title: DlOption<String>,
}

#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
pub struct TicketCopy {
    pub id: i64,
    pub title: DlOption<String>,
}

File: v6/prolog/compile/test/typegen_golden/generic_expansion_end_to_end.types.ts
export type Option<T> = { tag: 'none' } | { tag: 'some'; value: T };

export interface AuthorAudit {
  id: number;
  tag: string;
}

export interface Bucket {
  id: number;
}

export interface Metric {
  id: number;
  value: Option<number>;
}

export interface MetricCopy {
  id: number;
  value: Option<number>;
}

export interface Person {
  id: number;
  name: string;
}

export interface PriorityHigh {
  id: number;
}

export interface PriorityLow {
  id: number;
}

export interface PriorityTag {
  id: number;
  tag: string;
}

export interface Review {
  id: number;
}

export interface SeenBucket {
  id: number;
  list_id: number;
}

export interface SeenReview {
  id: number;
  name: string;
}

export interface SeenTicket {
  id: number;
  tag: string;
}

export interface Ticket {
  id: number;
  title: Option<string>;
}

export interface TicketCopy {
  id: number;
  title: Option<string>;
}
